classifiers/reference_quality/special_character_classifier/code_snippet_refinery.md

```diff
@@ -0,0 +1,27 @@
+```python
+import unicodedata
+from typing import Optional, List, Tuple
+
+ATTRIBUTE: str = "text" # only text attributes
+ALLOWED_RANGES: List[Tuple[int,int]] = None
+
+default_allowed_values = set(range(0x0020, 0x007F)).union( # Basic Latin
+    set(range(0x00A0, 0x00FF)), # Latin-1 Supplement
+    set(range(0x0100, 0x017F)),  # Latin Extended-A
+    set(range(0x0180, 0x024F)),  # Latin Extended-B
+    set(range(0x2000, 0x206F)),  # General Punctuation
+    set(range(0x20A0, 0x20CF)),  # Currency Symbols
+    set([ord("\t"), ord("\n"), ord("\r")])# common stop chars
+)  
+
+def special_character_classifier(record):
+    text = record[ATTRIBUTE].text    
+
+    allowed = ALLOWED_RANGES
+    if not allowed:
+        allowed = default_allowed_values
+    for char in text:
+        if ord(char) not in allowed and unicodedata.category(char) != "Zs":
+            return "true"
+    return "false"
```

Comment:
usually we use more descriptive labels and only for the true case. So maybe LABEL as variable and "contains special" as default?

```suggestion
        if ord(char) not in allowed and unicodedata.category(char) != "Zs":
            return LABEL
```

Reply:
Added this.